.github/workflows/ghci.yml

```diff
@@ -0,0 +1,146 @@
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15]
+        cc: [gcc, clang]
+        exclude:
+          - os: macos-10.15
+            cc: gcc
+        include:
+          # Enable distcheck for one build
+          - os: ubuntu-18.04
+            cc: gcc
+            do_distc: yes
+          # Run Coverity on a clang build; Coverity's gcc causes issues
+          - os: ubuntu-18.04
+            cc: clang
+            do_coverity: yes
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+
+      - name: retrieve caches
+        uses: actions/cache@v2
+        with:
+          path: |
+            $HOME/Library/Caches/Homebrew
+            /usr/local/Homebrew
+          key: ${{ runner.os }}-${{ matrix.os }}-build
+
+      - name: install deps
+        run: |
+          if (echo "${{ matrix.os }}" | grep -qE '^macos-') ; then
+            #brew update
+            # freetype, autoconf and libtool are preinstalled
+            # and `brew install` fails if a non-uptodate version is already installed
+            #brew upgrade freetype autoconf libtool
+            brew install automake fribidi harfbuzz nasm
+          else
+            #sudo apt-get update && sudo apt-get upgrade
+            sudo apt-get install -y \
+                 libfontconfig1-dev libfreetype6-dev libfribidi-dev \
+                 libharfbuzz-dev nasm ${{ matrix.cc }}
+          fi
+
+      - name: configure
+        run: ./autogen.sh && CC="${{ matrix.cc }}" ./configure
+
+      - name: compile
+        run: make -j 2
+
+      - name: distcheck
+        run: if [ "x${{ matrix.do_distc }}" == "xyes" ] ; then make -j 2 distcheck; fi
+
+      - name: Coverity scan
+        env:
+          COVERITY_SCAN_TOKEN: ${{ secrets.COVERITY_SCAN_TOKEN }}
+          PROJECT_NAME: libass/libass
+          NOTIFY_EMAIL: none@example.com
+          TOOL_URL: https://scan.coverity.com/download/
+          UPLOAD_URL: https://scan.coverity.com/builds?project=libass%2Flibass
+          SCAN_URL: https://scan.coverity.com
+          RES_DIR: cov-int
+        run: |
+          if [ "x${{ matrix.do_coverity }}" = "xyes" ] \
+             && [ "x${{ github.repository }}" = "xlibass/libass" ] \
+             && [ "x${{ github.event_name }}" != "xpull_request" ]
+          then
+            exit_code=0
+            echo "Running Coverity ..."
+            # Remove previous build output
+            make clean
+            # The upstream script is borked and always exits with 1 even on success
+            # To get meaningful success/error status we're using our own script
+            # but we still want to be informed about upstream script changes
+            if curl -s https://scan.coverity.com/scripts/travisci_build_coverity_scan.sh \
+                | shasum -a 256 \
+                | grep -Eq '^234d71b4a5257a79559e66dd3ba5765576d2af4845da83af4975b77b14ab536b '
+            then
+               echo "" > /dev/null
+            else
```

Comment:
Why not invert the condition with `if ! curl...` and drop the successful branch altogether?

Reply:
While `test ! …` is part of [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html) and [considered portable](https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Limitations-of-Builtins.html#test), `if !` apparently [is not](https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Limitations-of-Builtins.html#if). 
On the other hand, in this case we know the code runs in `bash` or `dash`, both of which support `if !`.